src/en/kunmangato/src/eu/kanade/tachiyomi/extension/en/kunmangato/KunMangaTo.kt

```diff
@@ -0,0 +1,267 @@
+    }
+
+    override fun searchMangaParse(response: Response): MangasPage {
+        val document = response.asJsoup()
+        val mangas = document.select(searchMangaSelector()).map { element ->
+            searchMangaFromElement(element)
+        }
+        val nextPageElement = document.select(".pagination .page-item").last()
+        return MangasPage(
+            mangas,
+            nextPageElement?.hasClass("disabled")?.not() ?: false,
+        )
+    }
+
+    override fun searchMangaSelector(): String {
+        return "article .card-manga"
+    }
+
+    override fun searchMangaFromElement(element: Element): SManga = popularMangaFromElement(element)
+
+    override fun latestUpdatesRequest(page: Int): Request {
+        val urlBuilder = "$baseUrl/latest".toHttpUrl().newBuilder()
+            .addQueryParameter("page", page.toString())
+
+        return GET(urlBuilder.build())
+    }
+
+    override fun latestUpdatesParse(response: Response): MangasPage {
+        val document = response.asJsoup()
+        val mangas = document.select(latestUpdatesSelector()).map { element ->
+            latestUpdatesFromElement(element)
+        }
+        val nextPageElement = document.select(".pagination .page-item").last()
+        return MangasPage(
+            mangas,
+            nextPageElement?.hasClass("disabled")?.not() ?: false,
+        )
+    }
+
+    override fun latestUpdatesSelector(): String = searchMangaSelector()
+
+    override fun latestUpdatesFromElement(element: Element) = searchMangaFromElement(element)
+
+    override fun mangaDetailsParse(document: Document): SManga {
+        return SManga.create().apply {
+            title = document.select(".page-heading").first()!!.text()
+            author = document.select("p.mb-1:nth-child(1)").first()!!.text().drop(9)
+            description = document.getElementById("manga-description")!!.text()
+            genre =
+                document.select("a.manga-genre").joinToString(", ") { element -> element.text() }
+            status =
+                if (document.select(".text-success").size > 0) SManga.COMPLETED else SManga.ONGOING
+            thumbnail_url = document.select("img.text-end").first()!!.attr("src")
+        }
+    }
+
+    override fun chapterListParse(response: Response): List<SChapter> {
+        val document = response.asJsoup()
+        val chaptersDocument =
+            Jsoup.parse("<div>" + document.getElementById("chapterList")!!.attr("value") + "</div>")
+        val chapterItems = chaptersDocument.select(".chapter-item")
+        return chapterItems.map { element ->
+            chapterFromElement(element)
+        }
+    }
+
+    override fun chapterFromElement(element: Element): SChapter {
+        val chapterName = element.getElementsByTag("h3").first()!!.text()
+
+        return SChapter.create().apply {
+            url = element.getElementsByTag("a").first()!!.attr("href").removePrefix(baseUrl)
+            name = chapterName
```

Comment:
why not just
```suggestion
            name = element.getElementsByTag("h3").first()!!.text()
```

Reply:
Because of what's next, cached the value, because it's used twice
> chapterName.removePrefix("Chapter ").toFloat()
Is it okay?

Reply:
In this case just remove the `chapter_number` field, the app will parse automatically